events: Do not generate reply fallbacks anymore :tada:

zecakeh opened the pull request:
According to [MSC2781](https://github.com/matrix-org/matrix-spec-proposals/pull/2781) / [Spec PR](https://github.com/matrix-org/matrix-spec/pull/1994).

The helper methods for the relevant relations have been simplified to only take the necessary data.

zecakeh committed:
events: Do not generate reply fallbacks anymore

According to MSC2781.

The helper methods for the relevant relations have been simplified to
only take the necessary data.

jplatte commented on crates/ruma-events/src/room/message.rs:
```diff
@@ -151,84 +150,45 @@ impl RoomMessageEventContent {
     }
 
-    /// Turns `self` into a reply to the given message.
+    /// Turns `self` into a [rich reply] to the message providing the given metadata.
```
```suggestion
    /// Turns `self` into a [rich reply] to the message using the given metadata.
```

zecakeh replied:
Applied

jplatte reviewed (approved):
Wait, we didn't merge this yet? Well, I guess that is good for some users 😂

Changelog needs updating now since the release is out, I think?

zecakeh commented:
Yes I didn't push this to be included in the release, because I don't know if it was wanted downstream. We can always backport not adding reply fallbacks in a non-breaking way in a patch release, if it's wanted.

zecakeh committed:
chore: Fix links to rich reply fallback section

Since they were remove in Matrix 1.13, the section was removed and the information is now in an info box.
